src/python/pants/backend/docker/target_types.py

```diff
@@ -30,8 +33,24 @@ class DockerBuildArgsField(StringSequenceField):
 
 class DockerImageSourceField(SingleSourceField):
     default = "Dockerfile"
+    default_glob_match_error_behavior = GlobMatchErrorBehavior.ignore
+    expected_num_files = range(0, 2)
+    required = False
+    help = (
+        "The Dockerfile to use when building the Docker image.\n\n"
+        "Use the `instructions` field instead if you prefer not having the Dockerfile in your "
+        "project source tree."
+    )
+
+
+class DockerImageInstructionsField(StringSequenceField):
+    alias = "instructions"
     required = False
-    help = "The Dockerfile to use when building the Docker image."
+    help = (
+        "The `Dockerfile` content, typically one instruction per list item.\n\n"
```

Comment:
A small example would help a lot, I think.

Per https://github.com/pantsbuild/pants/pull/13549, should use ``` to escape it for the docsite

Reply:
In markdown, I prefer to indent code blocks. So if you don't mind that for some stylistic reasons, I'll go with that. (it does work) See:
https://www.pantsbuild.org/v2.8/docs/reference-docker_image#coderegistriescode
https://github.com/pantsbuild/pants/blob/5798377b2876dc1d5355e407972a441ea7af06d7/src/python/pants/backend/docker/target_types.py#L64-L83

Reply:
Oh okay, given that it indeed does work on Readme, sounds good!